tests/e2e/fullsync_test_for_block_volume.go

```diff
@@ -769,4 +770,114 @@ var _ bool = ginkgo.Describe("full-sync-test", func() {
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
 
+	/*
+		Attach volume to a new pod when CNS is down and verify volume metadata in CNS post full sync
+		Steps:
+			1.	create a pvc pvc1, wait for pvc bound to pv
+			2.	create a pod pod1, using pvc1
+			3.	stop vsan-health on VC
+			4.	when vsan-health is stopped, delete pod1
+			5.	create a new pod pod2, using pvc1
+			6.	start vsan-health on VC
+			7.	after fullsync is triggered verify the CNS metadata for the volume backing pvc1
+			8.	delete pod2
+			9.	delete pvc1
+	*/
+	ginkgo.It("[csi-block-vanilla] [csi-supervisor] [csi-guest] [csi-block-vanilla-serialized] "+
+		"Attach volume to a new pod when CNS is down and verify volume metadata in CNS post full sync", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("create a pvc pvc1, wait for pvc bound to pv")
+		volHandle, pvc, pv, _ := createSCwithVolumeExpansionTrueAndDynamicPVC(
+			f, client, "", storagePolicyName, namespace)
+
```

Comment:
Add a defer function to delete SC, and PVC on failure

Reply:
Added